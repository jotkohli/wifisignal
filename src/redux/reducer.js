import {combineReducers} from 'redux';
import dataReducer from '../module/mainSaga/reducer';
const appReducer = combineReducers({
  data: dataReducer,
});

export default appReducer;
