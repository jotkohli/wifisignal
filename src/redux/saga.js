import {all} from 'redux-saga/effects';

import sagaData from '../module/mainSaga/saga';
export default function* rootSaga() {
  yield all([sagaData()]);
}
