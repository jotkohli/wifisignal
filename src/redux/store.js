import {configureStore} from '@reduxjs/toolkit';
import createSagaMiddleware from 'redux-saga';
import appReducer from './reducer';
import rootSaga from './saga';

const sagaMiddleware = createSagaMiddleware();
const middleware = [sagaMiddleware];

export const store = configureStore({
  reducer: appReducer,
  middleware: getDefaultMiddleware =>
    getDefaultMiddleware({
      serializableCheck: false,
    }).concat(middleware),
});

sagaMiddleware.run(rootSaga);
