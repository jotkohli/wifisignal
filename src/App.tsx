import React, {useEffect, useState} from 'react';
import {
  View,
  TextInput,
  Button,
  StyleSheet,
  Alert,
  Text,
  SafeAreaView,
  ScrollView,
} from 'react-native';

import {Provider, useDispatch, useSelector} from 'react-redux';

import {store} from './redux/store';
import Main from './main';
const App = () => {
  return (
    <Provider store={store}>
      <Main />
    </Provider>
  );
};

export default App;
