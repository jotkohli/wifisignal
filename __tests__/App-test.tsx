/**
 * @format
 */

import 'react-native';
import React from 'react';
import App from '../App';
import {render as rtlRender} from '@testing-library/react-native';
import {Provider} from 'react-redux';
import {store} from '../src/redux/store';

const render = (component: React.ReactElement) =>
  rtlRender(<Provider store={store}>{component}</Provider>);

it('renders correctly', () => {
  const {toJSON} = render(<App />);
  expect(toJSON()).toMatchSnapshot();
});
