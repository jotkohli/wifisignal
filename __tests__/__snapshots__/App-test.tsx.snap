// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<RCTSafeAreaView
  style={
    {
      "flex": 1,
    }
  }
>
  <View
    style={
      {
        "alignItems": "center",
        "alignSelf": "center",
        "flexDirection": "row",
        "marginBottom": 20,
        "width": "95%",
      }
    }
  >
    <Image
      source={
        {
          "testUri": "../../../src/images/search.png",
        }
      }
      style={
        {
          "height": 30,
          "marginLeft": 5,
          "position": "absolute",
          "tintColor": "grey",
          "width": 30,
        }
      }
    />
    <TextInput
      onChangeText={[Function]}
      placeholder="Enter user name"
      style={
        {
          "borderColor": "gray",
          "borderWidth": 1,
          "flex": 1,
          "height": 40,
          "marginRight": 10,
          "paddingLeft": 40,
        }
      }
      value=""
    />
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        [
          {
            "alignItems": "center",
            "backgroundColor": "#A7C941",
            "borderRadius": 18,
            "height": 40,
            "justifyContent": "center",
            "width": "25%",
          },
          {
            "opacity": 0.4,
          },
        ]
      }
    >
      <Text
        style={
          {
            "fontSize": 16,
          }
        }
      >
        Search
      </Text>
    </View>
  </View>
  <RCTScrollView>
    <View />
  </RCTScrollView>
</RCTSafeAreaView>
`;
